src/events/azure_sdk.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * @fileoverview Handlers for the azure-sdk js library instrumentation.
+ */
+const utils = require('../utils.js');
+const tracer = require('../tracer.js');
+const eventInterface = require('../event.js');
+const moduleUtils = require('./module_utils.js');
+
+/**
+ * Wraps the BlockBlobClient upload method.
+ * @param {Function} wrappedFunction The function to wrap
+ * @returns {Function} The wrapped function
+ */
+function blobUploadWrapper(wrappedFunction) {
+    return function internalUploadWrapper(content, size) {
+        const { accountName, containerName } = this;
+        const { slsEvent: uploadEvent, startTime } = eventInterface.initializeEvent(
+            'blob_storage',
+            containerName,
+            'upload',
+            'azure-sdk'
+        );
+        eventInterface.addToMetadata(uploadEvent, {
+            'azure.blob.account_name': accountName,
+            'azure.blob.container_name': containerName,
+            'azure.blob.content_size': size,
+        }, { 'azure.blob.content': content });
+        const request = wrappedFunction.apply(this, [content, size]);
+        const uploadPromise = new Promise((resolve) => {
+            request.then((res) => {
+                eventInterface.addToMetadata(uploadEvent, { 'azure.blob.error_code': res.errorCode });
+                uploadEvent.setDuration(utils.createDurationTimestamp(startTime));
+            }).catch((err) => {
+                eventInterface.setException(uploadEvent, err);
+            }).finally(() => {
+                resolve();
+            });
+        });
```

Comment:
if request is a promise, you should override it instead of wrapping with another promise (@idonava can point you to one of the recent instrumentations he did as an example)